`TestGetPeers` is flaky

## Flaky Test

### Which jobs are failing
```
=== RUN   TestGetPeers
    region_test.go:1326: 
        	Error Trace:	/home/runner/work/pd/pd/pkg/core/region_test.go:1326
        	Error:      	Not equal: 
        	            	expected: []*metapb.Peer{(*metapb.Peer)(0xc00004fc80), (*metapb.Peer)(0xc00004fc98)}
        	            	actual  : []*metapb.Peer{(*metapb.Peer)(0xc00004fc98), (*metapb.Peer)(0xc00004fc80)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,8 @@
        	            	 ([]*metapb.Peer) (len=2) {
        	            	+ (*metapb.Peer)({
        	            	+  Id: (uint64) 4,
        	            	+  StoreId: (uint64) 4,
        	            	+  Role: (metapb.PeerRole) 0,
        	            	+  IsWitness: (bool) false
        	            	+ }),
        	            	  (*metapb.Peer)({
        	            	@@ -6,8 +12,2 @@
        	            	   IsWitness: (bool) false
        	            	- }),
        	            	- (*metapb.Peer)({
        	            	-  Id: (uint64) 4,
        	            	-  StoreId: (uint64) 4,
        	            	-  Role: (metapb.PeerRole) 0,
        	            	-  IsWitness: (bool) false
        	            	  })
        	Test:       	TestGetPeers
--- FAIL: TestGetPeers (0.00s)
```

### CI link

https://github.com/tikv/pd/actions/runs/13696229199/job/38299167282

### Reason for failure (if possible)


### Anything else

--- a/pkg/core/region_test.go
+++ b/pkg/core/region_test.go
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"math"
 	mrand "math/rand"
+	"sort"
 	"strconv"
 	"testing"
 	"time"
@@ -1323,6 +1324,10 @@ func TestGetPeers(t *testing.T) {
 		},
 	} {
 		role := NewRole(v.role)
-		re.Equal(v.peers, region.GetPeersByRole(role))
+		peers := region.GetPeersByRole(role)
+		sort.Slice(peers, func(i, j int) bool {
+			return peers[i].Id <= peers[j].Id
+		})
+		re.Equal(v.peers, peers, role)
 	}
 }

